### PR TITLE
Basic tutorial

### LINKED ISSUE #5: Task 3: Beginner-friendly tutorial for “standard” single-cell analysis with scverse packages based on multiple tools
This should replace the “scanpy 3k PBMC” tutorial by a more modern workflow making use of multiple scverse packages, e.g. combining scanpy + scvi-tools + decoupler.

### PR DESCRIPTION
**Pull request by ivirshup**
Coauthored by: @dbdimitrov, @AnnaChristina, @Hrovatin 

## TODO before review:

- [ ] Actually add coauthors to commits
- [x] Figure out execution
- [ ] Complete notebook
- [ ] Test datasets

(restarting with branch on this repo to make docs render)

Close #5

### CODE REVIEW

**Comment by ivirshup** [comment]
I'm going to do a bad thing and add data to this PR

This is temporary until we figure out licensing for the data.

They can be read with:

```
import scanpy as sc, anndata as ad

adatas = {}

for dset in ["s1d1", "s1d3"]:
    sample_adata = sc.read_10x_h5(f"data/basic_scrna_tutorial/{dset}/filtered_feature_bc_matrix.h5")
    sample_adata.var_names_make_unique()
    adatas[dset] = sample_adata

adata = ad.concat(adatas, label="donor")
```

**Comment by AnnaChristina** [comment]
<p>no common consensus yet on scaling (no benchmark yet whether no scaling better recoveres valuable biological variations)</p><p> </p><p>I think would still be good to showcase how you could do it in scanpy, but agree can be an issue with dense matrices. Maybe we can add a warning or note to the notebook and only add this function as a commented-out? thoughts on this? </p>

---

 <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by AnnaChristina** [comment]
<p>Yes, wanted to actually expose target_sum=None instead of 1e6 to at least have this explaine. </p>

---

 <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by grst** [comment]
<p>what about this one? </p><p>https://www.biorxiv.org/content/10.1101/2021.06.24.449781v4.full</p><p> </p><p>(figure 2) </p><p> </p><p>(at least there's no big difference between scaling and not scaling) </p>

---

 <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by AnnaChristina** [comment]
<p>isn't this paper referring to the additional scaling step proposed by Booeshaghi et al. (https://www.biorxiv.org/content/10.1101/2022.05.06.490859v1.full) where they claim PFlog1pPF is the favourable normalization method? </p><p> </p><p>we can also remove it if you think easier to grasp for newbies or until there is a peer-reviewed version of one of these two normalization benchmarks. </p>

---

 <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by grst** [comment]
image

they say this doesn't work well

> We also advise against scaling the results of the shifted logarithm by the sum of the transformed values per cell as, e.g., suggested by [Booeshaghi et al. (2022)](https://www.biorxiv.org/content/10.1101/2021.06.24.449781v4.full#ref-8). In our hands ([Suppl. Fig. S1](https://www.biorxiv.org/content/10.1101/2021.06.24.449781v4.full#F6)), this additional operation failed to remove the confounding effect of the sequencing depth (the authors’ stated motivation for it) and did not improve the k-NN recall performance.

**Comment by grst** [comment]
Anyway, fair enough to show it as an optional step.

**Comment by dbdimitrov** [comment]
<p>Hmm I think many of these comments are now outdated (I essentially removed that part).</p>

---

 <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by dbdimitrov** [comment]
<p>Will do.</p>

---

 <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by dbdimitrov** [comment]
<p>It's the minimum genes per geneset (here cell type) to be considered</p><p> </p>

---

 <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by dbdimitrov** [comment]
<p>Will add a comment for it.</p>

---

 <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ivirshup** [comment]
<p>Just as a code style comment on this section, I would do:</p><p> </p><p><code></code><code>python</p><p>specificity_msk = markers['human_specificity'] &gt; 0.1</p><p>immune_markers = markers[immune_msk &amp; specificity_msk]</p><p></code><code></code></p><p> </p><p> </p>

---

 <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: docs/notebooks/basic-scrna-tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,1917 @@
+{
```
Comment (Zethson) [inline docs/notebooks/basic-scrna-tutorial.ipynb:1]: <p>Careful! You are linking to the surface protein aka CITE-seq markers here. You might need to clarify this.</p><p> </p><p>The actual RNA annotation chapter PR is here: https://github.com/theislab/single-cell-best-practices/pull/85 it is like 95% done and should hopefully get merged soon.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (namsaraeva) [inline docs/notebooks/basic-scrna-tutorial.ipynb:1]: <p>maybe a note on that "MT-" is for human and "Mt-" for mice? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (namsaraeva) [inline docs/notebooks/basic-scrna-tutorial.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    # hemoglobin genes
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (namsaraeva) [inline docs/notebooks/basic-scrna-tutorial.ipynb:1]: <p>obtained (typo)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='83'/>

**Comment by dbdimitrov** [comment]
The python environment that I was using:
https://gist.github.com/dbdimitrov/79483e31942dcfff439e060aaf1e934d

**Comment by grst** [comment]
Running into the first dependency issue

```
TypeError: np.matrix is not supported. Please convert to a numpy array with np.asarray. For more information see: https://numpy.org/doc/stable/reference/generated/numpy.matrix.html
```

somewhere in celltypist code. What do we need to pin?

File: docs/notebooks/basic-scrna-tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,1193 @@
...
```
Comment (grst) [inline docs/notebooks/basic-scrna-tutorial.ipynb:1]: <p>Could x-ref the new AnnData tutorial</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (grst) [inline docs/notebooks/basic-scrna-tutorial.ipynb:1]: <p>pls link packages and possible the benchmark paper or chapter from the best practice book</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (grst) [inline docs/notebooks/basic-scrna-tutorial.ipynb:1]: <p>Code below was switched from CPM to proportional fitting (PF). The text should be updated accordingly. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (grst) [inline docs/notebooks/basic-scrna-tutorial.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    # CPM count depth scaling
</pre><p>Comment not correct any more, this is proportional fitting.</p><p>Consider explicit <code>target_sum=None</code>  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (grst) [inline docs/notebooks/basic-scrna-tutorial.ipynb:1]: <p>Annotation -&gt; Cell-type annotation? </p><p> </p><p>I also started wondering if this should become a tutorial on it's own as it's quite comprehensive and shows multiple ways of doing so. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Comment (grst) [inline docs/notebooks/basic-scrna-tutorial.ipynb:1]: <p>typo: ultemately -&gt; ultimately</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Comment (grst) [inline docs/notebooks/basic-scrna-tutorial.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    adata.obs["manual_anno"] = [new_cluster_names[x] for x in adata.obs["leiden_res0_5"]]
</pre><p> </p><p> </p><pre class="ql-syntax" spellcheck="false">adata.obs["manual_anno"] = adata.obs["leiden_res0_5"].map(new_cluster_names)
</pre><p>would be the pandas way of doing this.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scverse-tutorials/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='60'/>

Comment (ivirshup) [reply docs/notebooks/basic-scrna-tutorial.ipynb:1]: <p>I've updated, but language could be reworded.</p>

**Comment by grst** [comment]
CI is passing now :) 
https://scverse-tutorials--21.org.readthedocs.build/en/21/notebooks/basic-scrna-tutorial.html#marker-gene-set

What is left to be done for the PR and who's gonna do it?

**Comment by grst** [comment]
> What is left to be done for the PR and who's gonna do it?

By reviewing I found mostly minor things. We still need a solution for the test dataset.

**Review by grst — APPROVED** [review]

**Pull request merged**